### ISSUE TITLE
[Load Balancer] Not exposing ports correctly on localhost

### ISSUE THREAD

**Issue by deadbeef84** [rancher/rancher#3683]
Rancher Version: 0.59.0

Docker Version: 1.9.1

OS: Ubuntu 15.10

Steps to Reproduce:
- Start rancher-server on localhost
- Start rancher-agent on localhost
- Start the following stack:
```
loadbalancer:
  ports:
  - 80:80
  tty: true
  image: rancher/load-balancer-service
  links:
  - web:web
  stdin_open: true
web:
  image: httpd
```
- Try to access loadbalancer on localhost:

Results:

```
$ curl http://localhost
curl: (56) Recv failure: Connection reset by peer
```

Expected: Successful response, as below.

It seems haproxy configuration binds directly to the rancher-ip:
`bind 10.42.57.243:80`

Using this IP directly works fine:
```
curl http://10.42.57.243`
<html><body><h1>It works!</h1></body></html>
```

**Comment by deadbeef84** [comment]
This might be a duplicate of #2291, but it should have been fixed as #1920 is now fixed/closed. I'm thinking the problem is the haproxy bind, and no longer hairpin nat.

**Comment by deniseschannon** [comment]
Sounds more like a duplicate of https://github.com/rancher/rancher/issues/3618 to me.
